tests/ci/gtest_util.sh

```diff
@@ -0,0 +1,26 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+function shard_gtest() {
+    export GTEST_TOTAL_SHARDS=$(nproc --all)
+    if [ -n "${2}" ]; then
+        GTEST_TOTAL_SHARDS="${2}"
+    fi
+    if [ -z ${GTEST_TOTAL_SHARDS} -o ${GTEST_TOTAL_SHARDS} -lt 1 ]; then
+        GTEST_TOTAL_SHARDS=4
+    fi
+
+    PIDS=()
+    COUNTER=0
+    while [ $COUNTER -lt $GTEST_TOTAL_SHARDS ]; do
+        export GTEST_SHARD_INDEX=$COUNTER
+        echo shard_gtest-Command: ${1}
```

Comment:
nit: this is invariant over each shard; should we just output it once before the loop?